styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -129,20 +139,35 @@
   }
 
   @Override
-  public String createExecution(final RunState runState, final String execName, final FlyteExecConf flyteExecConf,
-                                final Map<String, String> annotations)
+  public String createExecution(final RunState runState, final String execName, final FlyteExecConf flyteExecConf)
       throws CreateExecutionException {
     requireNonNull(runState, "runState");
     requireNonNull(execName, "name");
     requireNonNull(flyteExecConf, "flyteExecConf");
-    requireNonNull(annotations, "annotations");
     final var launchPlanIdentifier = flyteExecConf.referenceId();
     final var execMode = runState.data().trigger()
         .map(this::toFlyteExecutionMode)
         .filter(mode -> mode != ExecutionMode.UNRECOGNIZED)
         .orElseThrow(() -> new CreateExecutionException("Missing trigger or unknown in StateData: " + runState.data()));
 
-    var extraDefaultInputs = getExtraDefaultInputs(runState.workflowInstance(), runState.data());
+    var styxVariables = getStyxVariables(runState.workflowInstance(), runState.data());
+    var triggeredParams = runState.data().triggerParameters()
+        .map(FlyteAdminClientRunner::getFilteredTriggerParams)
+        .orElseGet(Map::of);
+
+    var labels = Map.of(STYX_EXECUTION_ID, normalize(styxVariables.get(STYX_EXECUTION_ID)));
+    final var annotations = ImmutableMap.<String, String>builder()
+        .putAll(styxVariables)
+        // just to keep compatibility with removing dangling executions
+        // TODO: base dangling logic on STYX_WORKFLOW_ID, STYX_PARAMETER and STYX_EXECUTION_ID
+        //   and the remove these annotations
+        .put(STYX_WORKFLOW_INSTANCE_ANNOTATION, styxVariables.get(STYX_WORKFLOW_ID) + "#" + styxVariables.get(STYX_PARAMETER))
```

Comment:
Would there be cases where `STYX_PARAMETER` in this format `2021-01-01T00:00:00`? Do we need to `normalize` the string as well?

Reply:
@tnsetting actually annotation values don't have a restriction on the format ([see](https://github.com/spotify/styx/pull/922)). Annotation keys they do have restrictions and in the past I mistakenly missread the restrictions for the keys as the restrictions for the values.

Following the link you will see an example that they use `"https://hub.docker.com/"` as value and you can notice `:` and `/`